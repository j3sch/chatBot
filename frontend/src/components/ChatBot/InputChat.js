import React, { useState } from 'react';
import { PaperAirplaneIcon } from '@heroicons/react/24/solid';

function Index({ sendMessage, isMobile }) {
	const [text, setText] = useState('');

	const MessageSend = () => {
		if (text) {
			sendMessage(text);
			setText('');
		}
	};

	const handleKeyDown = (event) => {
		if (event.key === 'Enter') {
			MessageSend();
		}
	};

	return (
		<div
			data-testid="id1"
			className="mt-8 bg-slate-100 dark:bg-slate-800 fixed bottom-0 right-0 z-50 "
			style={!isMobile ? { width: 'calc(100vw - 384px)' } : { width: '100vw' }}
		>
			<div className="flex flex-row justify-between items-center ml-2 pt-2 pb-2 ">
				<input
					id="chatInputField"
					className="w-full p-2 bg-white dark:bg-slate-600 dark:text-white rounded-lg border border-slate-300 dark:border-slate-800"
					value={text}
					onChange={(event) => {
						setText(event.target.value);
					}}
					onKeyDown={handleKeyDown}
				/>
				<div className="pl-4 pr-4 cursor-pointer" id="sendMessageButton" onClick={MessageSend}>
					<PaperAirplaneIcon
						className={`flex h-7 w-7 text-indigo-600 dark:text-white ${text ? '' : 'opacity-30'}`}
					/>
				</div>
			</div>
		</div>
	);
}

export default Index;
