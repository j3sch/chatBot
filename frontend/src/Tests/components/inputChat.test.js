import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import InputChat from '../../components/ChatBot/InputChat';

afterEach(() => {
	cleanup();
});

test('InputChat should be mobile', () => {
	render(<InputChat sendMessage={null} isMobile={true} />);
	const element = screen.getByTestId('id1');
	expect(element).toBeInTheDocument();
	expect(element).toHaveStyle({ width: '100vw' });
});

test('InputChat should not be mobile', () => {
	render(<InputChat sendMessage={null} isMobile={false} />);
	const element = screen.getByTestId('id1');
	expect(element).toBeInTheDocument();
	expect(element).toHaveStyle({ width: 'calc(100vw - 384px)' });
});

test('matches snapshot', () => {
	const tree = renderer.create(<InputChat sendMessage={null} isMobile={false} />).toJSON();
	expect(tree).toMatchSnapshot();
});
