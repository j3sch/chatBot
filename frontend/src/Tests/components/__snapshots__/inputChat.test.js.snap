// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="mt-8 bg-slate-100 dark:bg-slate-800 fixed bottom-0 right-0 z-50 "
  data-testid="id1"
  style={
    Object {
      "width": "calc(100vw - 384px)",
    }
  }
>
  <div
    className="flex flex-row justify-between items-center ml-2 pt-2 pb-2 "
  >
    <input
      className="w-full p-2 bg-white dark:bg-slate-600 dark:text-white rounded-lg border border-slate-300 dark:border-slate-800"
      id="chatInputField"
      onChange={[Function]}
      onKeyDown={[Function]}
      value=""
    />
    <div
      className="pl-4 pr-4 cursor-pointer"
      id="sendMessageButton"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="flex h-7 w-7 text-indigo-600 dark:text-white opacity-30"
        fill="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3.478 2.405a.75.75 0 00-.926.94l2.432 7.905H13.5a.75.75 0 010 1.5H4.984l-2.432 7.905a.75.75 0 00.926.94 60.519 60.519 0 0018.445-8.986.75.75 0 000-1.218A60.517 60.517 0 003.478 2.405z"
        />
      </svg>
    </div>
  </div>
</div>
`;
